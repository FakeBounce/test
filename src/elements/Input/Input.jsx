import React, { Component } from "react";
import styles from "./Input.module.scss";
import { Input } from "reactstrap";
import { observer } from "mobx-react";
import { observable } from "mobx";
import classNames from "classnames";

@observer
class InputElement extends Component {
  @observable inputValue = "";

  constructor() {
    super();
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  componentDidMount() {
    const { initialValue } = this.props;
    this.inputValue = initialValue ? initialValue : "";
  }

  componentWillReceiveProps(nextProps, nextContext) {
    if (nextProps.initialValue) {
      this.inputValue = nextProps.initialValue ? nextProps.initialValue : "";
    }
  }

  handleInputChange(event) {
    const { inputValue } = this.props;
    inputValue(event.target.value);
    this.inputValue = event.target.value;
  }

  render() {
    const { placeholder, className } = this.props;

    const containerClass = classNames(styles.InputContainer, className);

    return (
      <div className={containerClass || null}>
        <Input
          placeholder={placeholder}
          value={this.inputValue}
          onChange={this.handleInputChange}
        />
      </div>
    );
  }
}

export default InputElement;
