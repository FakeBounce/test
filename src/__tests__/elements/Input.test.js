import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import Input from "elements/Input/Input";

configure({ adapter: new Adapter() });
const handleInput = jest.fn();

describe("Testing Input component", () => {
  it("renders as expected", () => {
    const wrapper = shallow(<Input />);
    expect(wrapper).toMatchSnapshot();
  });
  it("functions behave correctly", () => {
    const wrapper = shallow(<Input inputValue={handleInput} />);
    wrapper.instance().handleInputChange({ target: { value: "test" } });
    expect(wrapper.instance().inputValue).toBe("test");
    expect(handleInput.mock.calls.length).toBe(1);
  });
});
