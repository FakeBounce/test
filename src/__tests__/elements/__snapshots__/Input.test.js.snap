// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputElement />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Input
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value=""
/>,
      "className": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onChange": [Function],
        "placeholder": undefined,
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Input
          onChange={[Function]}
          placeholder={undefined}
          type="text"
          value=""
/>,
        "className": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onChange": [Function],
          "placeholder": undefined,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
